<launch>
  <!-- Single-node white-line follower -->
  
<node pkg="line_detection"
      type="line_detection_node.py"
      name="white_line_detector_follower"
      ns="paul"
      output="screen">
      
    <!-- Subscribe to the real camera topic under your vehicle namespace -->
    <remap from="~image/compressed" to="/paul/camera_node/image/compressed"/>
    <!-- Publish commands where the switch node listens -->
    

    <param name="drive_mode" value="auto"/>
    <!-- White-line HSV thresholds -->
    <rosparam param="hsv_lower">[0, 0, 230]</rosparam>
    <rosparam param="hsv_upper">[180, 30, 255]</rosparam>

    <!-- Region-of-interest cropping -->
    <param  name="roi_fraction" value="0.5"/>
    <param  name="top_cutoff"   value="0"/>

    <!-- Hough-line detection settings -->
    <param name="hough_rho"       value="1"/>
    <param name="hough_theta"     value="0.0174533"/>
    <param name="hough_threshold" value="50"/>
    <param name="hough_min_line"  value="50"/>
    <param name="hough_max_gap"   value="20"/>

    <!-- Control gains & speeds -->
    <param name="Kp_pos"     value="0.005"/>
    <param name="Kp_ang"     value="1"/>
    <param name="v_straight" value="0.5"/>
    <param name="v_turn"     value="0.5"/>
    <param name="err_thresh" value="20.0"/>

    <!-- Enable debug -->
    <param name="debug" value="true"/>
  </node>
</launch>
